Automattic/wp-calypso
# Use WordPress / Gutenberg style lint guidelines for Calypso

### Summary
- Copy Gutenberg's CSS style guidelines exactly, while adding back our own allowed property rules.
- Use WordPress style guidelines for CSS -- including _**double** quotes everywhere_ and _**no** spaces within parentheses_.
- Disable `declaration-property-unit-allowed-list` within functions so that our `rem(20px)` function is not considered bad style.
- Ignore our build artifacts from stylelint.

### Proposed Changes
Work towards #67467. The main motivation for the issue and this PR is the number of new style warnings on trunk after merging #66909, making the DevX for coding styles less pleasant. 

In #66909, we made some progress in using the WordPress scss preset for our style guidelines. In this PR, I take the changes even further -- I've effectively copy/pasted https://github.com/WordPress/gutenberg/blob/trunk/.stylelintrc.json, and then added back the following rules:

- declaration-property-unit-allowed-list
- scales/font-weights
- scales/radii
- unit-allowed-list

The major change over the current trunk is that we no longer extend `stylelint-config-standard-scss`. I don't think we needed to add that in #66909, as it's introduced a few rules I don't think we care about following.

The major change over the _previous trunk_ (e.g. before #66909) is that we're [more closely matching the WordPress style guidelines](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/) in a few cases:
- We should be using double quotes, but currently use single. 
- We should _not_ have space inside parentheses for css.

There are potentially other changes, but those were the two major ones I noted. 

Additionally, I took the opportunity to ignore various artifacts to make the linter faster. It was previously linting node_modules and other build artifacts. We can't just point it at `.gitignore` because we have multiple `.gitignore`s in the repo, and stylelint only supports one ignore file.

#### Testing Instructions
Run `yarn lint:css --fix` locally.

Check the following things _in general_ after that completes:
- Are the rules it complains about rules we should care about? (If not, we should further tweak the rules.)
- Are the rules it fixes what we expect?
- Are there missing rules we should be enforcing?

I'm not expecting us to catch all of the above, but I am expecting this PR to improve the linting situation after merging, and after merging an autofix PR. I'm not expecting this PR to introduce further issues.
